modules/zenith-public/lua/3-complete/explorer_moogle.lua

```diff
@@ -0,0 +1,45 @@
+-- -----------------------------------
+-- Explorer Moogle Module
+--
+-- This module overrides the conditions to use explorer moogle teleports.
+-- -----------------------------------
+
+require('modules/module_utils')
+
+local m = Module:new('explorer_moogle')
+
+local noRank3Nation = function(player)
+    return
+        player:getRank(player:getNation(xi.nation.BASTOK)) < 3 and
+        player:getRank(player:getNation(xi.nation.SANDORIA)) < 3 and
+        player:getRank(player:getNation(xi.nation.WINDURST)) < 3
+end
+
+local canNotTeleport = function(player)
+    return noRank3Nation(player) and player:getMainLvl() >= xi.settings.main.EXPLORER_MOOGLE_LV
+end
```

Comment:
```suggestion
local cannotTeleport = function(player)
    local noRank3Nation = player:getRank(player:getNation(xi.nation.BASTOK)) < 3 and
        player:getRank(player:getNation(xi.nation.SANDORIA)) < 3 and
        player:getRank(player:getNation(xi.nation.WINDURST)) < 3

    return noRank3Nation or player:getMainLvl() < xi.settings.main.EXPLORER_MOOGLE_LV
end
```

It's late and I'm willing to be shown to be wrong, but I think your logic would allow low level players to teleport?

Reply:
In this case I believe you are. In the original `explorerMoogleOnTrigger` there is already a check that increments the event if the player is under level 10. If we also increment the event if they are under level 10 then when we call `super()` it will be incremented twice. We don't need to handle player being under the level requirement. We need to handle when the player is over or equal to but not high enough rank.

```
xi.teleport.explorerMoogleOnTrigger = function(player, event)
    local accept = 0

    if player:getGil() < 300 then
        accept = 1
    end

*** This line here handles incrementing the event if they aren't high enough level. We don't want to duplicate it ***
    if player:getMainLvl() < xi.settings.main.EXPLORER_MOOGLE_LV then
        event = event + 1
    end

    player:startEvent(event, player:getZoneID(), 0, accept)
end
```

So in the case when they are under the level the `super()` handles that. In the case they are equal to or over but under the rank we handle that. If they meet both the only other check is if they have enough gil to pay.

Reply:
Oof confusing. Ok makes sense